tpot/config/classifier_cuml.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+"""This file is part of the TPOT library.
+
+TPOT was primarily developed at the University of Pennsylvania by:
+    - Randal S. Olson (rso@randalolson.com)
+    - Weixuan Fu (weixuanf@upenn.edu)
+    - Daniel Angell (dpa34@drexel.edu)
+    - and many more generous open source contributors
+
+TPOT is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+TPOT is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with TPOT. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+
+import numpy as np
+
+# This configuration provides users with access to a GPU the ability to
+# use cuML classifiers as estimators alongside the scikit-learn
+# preprocessors in the TPOT default configuration.
+
+classifier_config_cuml = {
+    # cuML Classifiers
+    
+    'cuml.neighbors.KNeighborsClassifier': {
+        'n_neighbors': range(1, 101),
+        'weights': ["uniform",],
+    },
+    'cuml.svm.SVC': {
+        'tol': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1,],
+        'C': [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1., 5., 10., 15., 20., 25.,]
+    },
+    "cuml.ensemble.RandomForestClassifier": {
+        'n_estimators': [100, 300, 500,],
+        'split_algo': [0, 1,],
+        'max_depth': range(8, 20),
+        'max_features': np.arange(0.05, 1.01, 0.05),
+        'min_rows_per_node': range(2, 21),
+        'n_bins': [8, 64,]
+    },
+    'cuml.linear_model.LogisticRegression': {
+        'penalty': ["l1", "l2", "elasticnet"],
+        'C': [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1., 5., 10., 15., 20., 25.,],
+    },
+    
+    # Sklearn Preprocesssors
+    
+    'sklearn.preprocessing.Binarizer': {
+        'threshold': np.arange(0.0, 1.01, 0.05)
+    },
+
+    'sklearn.decomposition.FastICA': {
+        'tol': np.arange(0.0, 1.01, 0.05)
+    },
+
+    'sklearn.cluster.FeatureAgglomeration': {
+        'linkage': ['ward', 'complete', 'average'],
+        'affinity': ['euclidean', 'l1', 'l2', 'manhattan', 'cosine']
+    },
+
+    'sklearn.preprocessing.MaxAbsScaler': {
+    },
+
+    'sklearn.preprocessing.MinMaxScaler': {
+    },
+
+    'sklearn.preprocessing.Normalizer': {
+        'norm': ['l1', 'l2', 'max']
+    },
+
+    'sklearn.kernel_approximation.Nystroem': {
+        'kernel': ['rbf', 'cosine', 'chi2', 'laplacian', 'polynomial', 'poly', 'linear', 'additive_chi2', 'sigmoid'],
+        'gamma': np.arange(0.0, 1.01, 0.05),
+        'n_components': range(1, 11)
+    },
+
+    'sklearn.decomposition.PCA': {
```

Comment:
How about using [PCA](https://github.com/rapidsai/cuml/blob/branch-0.15/python/cuml/decomposition/pca.pyx#L111) from cuml?

Reply:
Absolutely, that will work cleanly here as well. I'll include it in the next push that includes two example notebooks (one for classification and one for regression).

Reply:
Resolved by https://github.com/EpistasisLab/tpot/pull/1108/commits/5cea7b29f7093613972308d4c56f47bf60e80410

Reply:
Switched PCA back to scikit learn after testing and seeing slightly better results.